src/tracing/writer.rs

```diff
@@ -265,32 +265,38 @@ impl<W: Write> TraceWriter<W> {
             write!(
                 self.writer,
                 "{trace_kind_style}{CALL}new{trace_kind_style:#} {label}@{address}",
-                label = trace.decoded.label.as_deref().unwrap_or("<unknown>")
+                label =
+                    trace.decoded.as_ref().and_then(|d| d.label.as_deref()).unwrap_or("<unknown>")
             )?;
             if self.config.write_bytecodes {
                 write!(self.writer, "({})", trace.data)?;
             }
         } else {
-            let (func_name, inputs) = match &trace.decoded.call_data {
-                Some(DecodedCallData { signature, args }) => {
-                    let name = signature.split('(').next().unwrap();
-                    (name.to_string(), args.join(", "))
-                }
-                None => {
-                    if trace.data.len() < 4 {
-                        ("fallback".to_string(), hex::encode(&trace.data))
-                    } else {
-                        let (selector, data) = trace.data.split_at(4);
-                        (hex::encode(selector), hex::encode(data))
+            let (func_name, inputs) =
+                match &trace.decoded.as_ref().and_then(|d| d.call_data.as_ref()) {
```

Comment:
I think we can introduce some helper functions to access these fields than this becomes less verbose and can get rid of a bunch of nested and_then asref